src/components/MyKiva/BadgeCompleted.vue

```diff
@@ -0,0 +1,167 @@
+					:alt="badgeTitle"
+				>
+			</div>
+			<h2 class="tw-italic tw-font-medium tw-text-desert-rose-4 tw-mb-2 tw-text-center">
+				{{ funFact }}<a
+					:href="learnMoreLink"
+					v-kv-track-event="[
+						'portfolio',
+						'click',
+						'Already earned badge modal - Learn more',
+						badgeCategory,
+						badgeLevel
+					]"
+					class="tw-underline tw-text-desert-rose-4 hover:tw-text-desert-rose-4"
+				> Learn more</a>
+			</h2>
+			<h4 class="tw-text-secondary">
+				{{ earnedDate }}
+			</h4>
+		</div>
+		<KvSocialShareButton
+			style="max-width: 328px; margin: 0 auto"
+			v-kv-track-event="[
+				'portfolio',
+				'click',
+				'Already earned badge modal - Share badge',
+				badgeCategory,
+				badgeLevel
+			]"
+			variant="primary"
+			modal-title="Celebrate your badge!"
+			:share-message="shareMessage"
+			:share-url="shareUrl"
+			:utm-campaign="utmCampaign"
+			:utm-content="utmContent"
+		>
+			<div class="tw-flex tw-gap-1 tw-items-center">
+				<KvMaterialIcon :icon="mdiExportVariant" class="tw-w-2.5 tw-h-2.5 tw-text-white" />
+				<span class="tw-font-medium">Share</span>
+			</div>
+		</KvSocialShareButton>
+		<p class="tw-text-small tw-text-center tw-text-secondary tw-mt-1.5">
+			{{ funFactSource }}
+		</p>
+	</div>
+</template>
+
+<script setup>
+
+import KvSocialShareButton from '#src/components/Kv/KvSocialShareButton';
+import AnimatedStars from '#src/components/Thanks/AnimatedStars';
+import KvMaterialIcon from '@kiva/kv-components/vue/KvMaterialIcon';
+
+import confetti from 'canvas-confetti';
+import {
+	defineProps,
+	computed,
+	toRefs,
+	onMounted
+} from 'vue';
+import { format } from 'date-fns';
+import { mdiExportVariant } from '@mdi/js';
+
+const props = defineProps({
+	badge: {
+		type: Object,
+		default: () => ({}),
+	},
+	lender: {
+		type: Object,
+		default: () => ({}),
+	},
+	lendingAchievement: {
+		type: Object,
+		default: () => ({}),
+	},
+});
+
+const { badge, lender, lendingAchievement } = toRefs(props);
+
+const isPublic = computed(() => lender.value?.public && lender.value?.publicName);
+const shareUrl = computed(() => (isPublic.value ? `/lender/${lender.value?.publicId}` : 'https://www.kiva.org'));
+// eslint-disable-next-line max-len
+const shareMessage = "It's not everyday you change a life! Thank you, from all of us at Kiva and the millions of lives changed around the world.";
+
+const utmCampaign = computed(() => `social_share_portfolio_badge_${badge.value.id}`);
+const utmContent = computed(() => {
+	if (lender.value?.public && lender.value?.publicName) return lender.value.publicName;
+	return 'anonymous';
+});
+
+const badgeImage = computed(() => {
+	return badge.value.fields?.badgeImage?.fields?.file?.url ?? '';
+});
+
+const badgeCategory = computed(() => badge.value?.challengeName ?? '');
+const tiers = computed(() => lendingAchievement.value?.tiers ?? []);
+
+const currentTier = computed(() => {
+	return tiers.value?.findLast(tier => tier.completedDate) ?? null;
```

Comment:
The tiers aren't sorted, so you'll need to sort by tier target as well. Targets increase in size for the different tiers.